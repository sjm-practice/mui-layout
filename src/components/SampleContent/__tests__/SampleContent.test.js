/* eslint-env jest */

import React from "react";
import { render } from "@testing-library/react";

import SampleContent from "../index";

describe("<SampleContent />", () => {
  it("Should match Snapshot", () => {
    const { asFragment } = render(<SampleContent />);

    expect(asFragment()).toMatchSnapshot();
  });
});
