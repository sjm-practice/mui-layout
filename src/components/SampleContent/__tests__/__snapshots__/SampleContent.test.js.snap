// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SampleContent /> Should match Snapshot 1`] = `
<DocumentFragment>
  <div
    class="ContentEx-root-1"
  >
    <span
      class="MuiTypography-root MuiTypography-overline"
    >
      INTRODUCING
    </span>
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
      weight="bold"
    >
      Material UI 
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
      >
        Layout
      </a>
    </h4>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
    >
      <b>
        Version 1.1
      </b>
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1"
      indent="small"
    >
      Layout is a group of Material-UI components that are enhanced and combined to create dynamic, easy-to-maintain and easy-to-code as much as possible.
    </p>
    <br />
    <br />
    <br />
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      weight="bold"
    >
      Let's start with Why?
    </h5>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
    >
      I created this because
      <br />
    </p>
    <div
      class="MuiTypography-root MuiTypography-body1"
    >
      <ol>
        <li>
          It took me a lot of time to initialize dashboard layout when I have new projects and I’m sure that a lot of you guys agree with me.
        </li>
        <li>
          Sometimes it is hard to refactor because the structure is so poor because someone isn’t deeply understand what he/she was doing, as a result, rewrite the whole layout (it actually happened, at least in my experience).
        </li>
        <li>
          Because we need to be fast to let others continue our work, we frequently write poor and a lot of code. However, we say we don’t have time to fix them. Eventually, spend all day paying technical debts.
        </li>
      </ol>
    </div>
    <br />
    <br />
    <br />
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      weight="bold"
    >
      Objectives
    </h5>
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      It must be easy enough to use, however still be able to adjust to compat with real word examples and usages. More importantly, it need to follow 
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary"
        href="https://material.io/design/components/navigation-drawer.html#"
        rel="noopener"
        target="_blank"
      >
        Material specs
      </a>
       since we are 100% based on Material-UI with no other styling libraries. Last but not least, responsive is a must.
    </p>
    <br />
    <br />
    <br />
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      weight="bold"
    >
      Solution
    </h5>
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      Separate layout into 5 components
    </p>
    <div
      class="MuiTypography-root MuiTypography-body1"
    >
      <ul>
        <li>
          Root
        </li>
        <li>
          Header (AppBar)
        </li>
        <li>
          Nav (Drawer)
        </li>
        <li>
          Content
        </li>
        <li>
          Footer
        </li>
      </ul>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      <b>
        Root
      </b>
       will provide context to other components to sync states across them.
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      <b>
        Behavior
      </b>
       of your layout will be controlled by using config (just a plain object) that will be injected to the Root.
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      <b>
        Presets
      </b>
       is a set of predefined config that I can come up with after researching a lot of the real world websites and also the 
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="https://material.io/design/components/navigation-drawer.html#usage"
      >
        official Material specs
      </a>
    </p>
    <br />
    <br />
    <br />
  </div>
</DocumentFragment>
`;
